lib/library.go

```diff
@@ -210,139 +212,59 @@ func Logout() *C.char {
 	return makeJSONResponse(err)
 }
 
-//ApproveSignRequestWithArgs instructs backend to complete sending of a given transaction.
-// gas and gasPrice will be overrided with the given values before signing the
-// transaction.
-//export ApproveSignRequestWithArgs
-func ApproveSignRequestWithArgs(id, password *C.char, gas, gasPrice C.longlong) *C.char {
-	result := statusBackend.ApproveSignRequestWithArgs(C.GoString(id), C.GoString(password), int64(gas), int64(gasPrice))
-
-	return prepareApproveSignRequestResponse(result, id)
-}
-
-//ApproveSignRequest instructs backend to complete sending of a given transaction.
-//export ApproveSignRequest
-func ApproveSignRequest(id, password *C.char) *C.char {
-	result := statusBackend.ApproveSignRequest(C.GoString(id), C.GoString(password))
-
-	return prepareApproveSignRequestResponse(result, id)
-}
-
-// prepareApproveSignRequestResponse based on a sign.Result prepares the binding
-// response.
-func prepareApproveSignRequestResponse(result sign.Result, id *C.char) *C.char {
-	errString := ""
-	if result.Error != nil {
-		fmt.Fprintln(os.Stderr, result.Error)
-		errString = result.Error.Error()
-	}
-
-	out := SignRequestResult{
-		ID:    C.GoString(id),
-		Hash:  result.Response.Hex(),
-		Error: errString,
-	}
-	outBytes, err := json.Marshal(out)
+// SignMessage unmarshals rpc params {data, address, password} and passes
+// them onto backend.SignMessage
+// nolint: deadcode
+func SignMessage(rpcParams *C.char) *C.char {
```

Comment:
I would extract the password to a separate argument. It will allow status-react to pass params directly from the caller and just send password as a next argument. With this approach, they will need to change callers params by extending it with a password. You can talk with someone from Clojure Team to clarify that.

Reply:
@flexsurfer What do you think about this interface vs adding the password as a separate argument as adam is suggesting?

Reply:
for us passing password as a  separate argument is more convenient , thanks